### PR TITLE
Add IQueueRepository

### PR DESCRIPTION
**Pull request by matee911**

### CODE REVIEW

File: src/ClayPipe/Infrastructure/QueueRepository.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+using System.Collections.Generic;
+using ClayPipe.Models;
+
+namespace ClayPipe.Infrastructure
+{
+    class QueueRepository : IQueueRepository
```
Comment (ekamil) [inline src/ClayPipe/Infrastructure/QueueRepository.cs]: Proposition - move it to test catalogue and rename QueueRepositoryMock.
Will be useful in unit tests.

File: src/ClayPipe/Models/QueueRepository.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+
+namespace ClayPipe.Models
+{
+    public interface IQueueRepository
+    {
+        IEnumerable<DeadLetterMessage> Pop();
+        void Put(DeadLetterMessage dlm);
+        void Put(params DeadLetterMessage[] dlm);
+        IEnumerable<QueueStats> Stats();
```
Comment (ekamil) [inline src/ClayPipe/Models/QueueRepository.cs]: Add a method: setQueueName, setVHost

File: src/ClayPipe/Models/QueueStats.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+namespace ClayPipe.Models
+{
+    public class QueueStats
+    {
+        public readonly string name;
+        public readonly string vHost;
+        public readonly int messagesCount;
+        public QueueStats(string name, string vHost, int messagesCount)
```
Comment (ekamil) [inline src/ClayPipe/Models/QueueStats.cs]: Blank line before pls :)

**Comment by ekamil** [comment]
Fixes #14

File: src/ClayPipe/Infrastructure/Tests/QueueRepositoryMock.cs [hunk 4]
```diff
@@ -3,8 +3,21 @@
 
 namespace ClayPipe.Infrastructure.Tests
 {
+    public class MockQueueRepositoryFactory : IQueueRepositoryFactory
```
Comment (matee911) [inline src/ClayPipe/Infrastructure/Tests/QueueRepositoryMock.cs]: MockFooBar or FooBarMock?

Comment (ekamil) [inline src/ClayPipe/Infrastructure/Tests/QueueRepositoryMock.cs]: MockFooBar. See 0874ca7

**Pull request merged**